jms/src/main/scala/akka/stream/alpakka/jms/Jms.scala

```diff
@@ -15,6 +17,19 @@ sealed trait Destination
 final case class Topic(name: String) extends Destination
 final case class Queue(name: String) extends Destination
 
+final case class JmsTextMessage(body: String, properties: Map[String, Any] = Map.empty) {
```

Comment:
The reason is mainly due to the fact that in order to create a JMS message you need to do it via the JmsSession 
```
val textMessage: TextMessage = jmsSession.session.createTextMessage(elem.body)
```
and I wanted to be able to pass JMS message properties. 

I am not happy with the solution in part because the API is currently not symmetric: `Sink[akka.stream.alpakka.jms.JmsTextMessage, NotUsed]` vs. `Source[javax.jms.Message, NotUsed]`. Furthermore the new type currently just adds a convenience method to add JMS message properties. 

We should either get rid of `akka.stream.alpakka.jms.JmsTextMessage` or refactor `JmsSource` to use the new type `akka.stream.alpakka.jms.JmsTextMessage`. I am inclined to go for the first option (getting rid of `akka.stream.alpakka.jms.JmsTextMessage`), but for this we would need to refactor the `JmsConnector` (see related #199).

Any suggestions?